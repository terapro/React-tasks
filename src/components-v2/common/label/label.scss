.lbl {
  height: $elementHeight;
  line-height: $elementHeight;
  font-size: 14px;
  color: $color_black;
  user-select: none;
}

.lbl-UC {
  text-transform: uppercase;
}
.lbl-white {
  color: $color_white;
}
.lbl-bold {
  font-weight: bold;
}