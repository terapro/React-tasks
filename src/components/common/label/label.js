import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Label = (props) => {

    const {labelUC, labelWhite, labelBold, content} = props;
    const itemClass =classNames({
      'lbl': true,
      'lbl-UC': labelUC,
      'lbl-white': labelWhite,
      'lbl-bold': labelBold,
    });

    return (
      <div className={itemClass}>
        {content}
      </div>
    )
};

Label.propTypes = {
    labelBold: PropTypes.bool,
    content: PropTypes.string,
    labelUC: PropTypes.bool,
    labelWhite: PropTypes.bool
};

Label.defaultProps = {
  labelBold: false,
  content: 'Label',
  labelUC: false,
  labelWhite: false
};

export {Label};
