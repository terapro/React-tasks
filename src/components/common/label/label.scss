.label {
  height: $elementHeight;
  line-height: $elementHeight;
  font-size: 14px;
  color: $color_black;
  user-select: none;
}

.label-upper-case {
  text-transform: uppercase;
}
.label-white {
  color: $color_white;
}