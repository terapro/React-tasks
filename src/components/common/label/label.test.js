import React from 'react';
import {Label} from 'src/components/common/label/label.js';
import {shallow} from 'enzyme';

const mockContent = 'Testing label';

describe('<Label />', () => {
  it('should render the label', () => {
    const wrapper = shallow(<Label content={mockContent} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should be configurable', () => {
    expect(shallow(<Label labelWhite />).is('.lbl-white')).toBe(true);
    expect(shallow(<Label labelBold />).is('.lbl-bold')).toBe(true);
  });
});
