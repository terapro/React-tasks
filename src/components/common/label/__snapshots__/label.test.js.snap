// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Rendering should render the label 1`] = `
<div
  className="lbl"
>
  Testing label
</div>
`;

exports[`<Label /> should render the label 1`] = `
<div
  className="lbl"
>
  Testing label
</div>
`;
